### ISSUE TITLE
Terminal run Git command can be "corrupted" if there is previous text waiting in the terminal

### ISSUE THREAD

**Issue by fterradev** [gitkraken/vscode-gitlens#2664]
### Description

On a macOS Monterey (12.6.3)

1. Right-click a commit in the commits panel
2. Select option "Reset Current Branch to Commit..."
3. Select option "Hard Reset"

Expect result:
The reset command works and the branch gets reset to the selected commit.

Actual result:
The reset command fails.
```
% git checkoutgit -C "/Users/johndoe/my-app" -c "core.editor=code --wait --reuse-window" reset --hard 9899d3cbb666bdb4e588cec6372f52b83bcfe656
git: 'checkoutgit' is not a git command. See 'git --help'.
```

Workaround: removing keyword 'checkoutgit' from the command seems to fix the issue.

### GitLens Version

13.5.0

### VS Code Version

Version: 1.78.0 (Universal)
Commit: 252e5463d60e63238250799aef7375787f68b4ee
Date: 2023-05-03T20:11:00.813Z
Electron: 22.4.8
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin arm64 21.6.0
Sandboxed: No

### Git Version

git version 2.37.1 (Apple Git-137.1)

**Comment by eamodio** [comment]
@fterradev The issue here was that it looks like you typed "git checkout" in the GitLens terminal (but never hit enter), and then want GL executed the reset command, it tacked it on to the existing "git checkout" text.

GitLens will now remove/overwrite any existing input text in the terminal.

**Comment by eamodio** [comment]
Can you please verify this fix in tomorrow's (`v2023.5.09xx`) pre-release edition of GitLens? Thank you so much!

You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/228608740-6071bc19-7b1a-44c8-843f-d2af44d6114b.png)

**Comment by fterradev** [comment]
> 

Oh, ok, I didn't realized it. Yes, I just reproduced it by typing something in the terminal. Thanks!
Didn't test the new version yet.

**Comment by fterradev** [comment]
> Can you please verify this fix in tomorrow's (`v2023.5.09xx`) pre-release edition of GitLens? Thank you so much!
> 
> You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.
> 
> ![image](https://user-images.githubusercontent.com/641685/228608740-6071bc19-7b1a-44c8-843f-d2af44d6114b.png)

Hey, I tested it didn't work...
Actually there seems to be a new defect.

It seems like the `git` command is now instead called as `usr/bin/git`. And I get something similar to this:
```
% usr/bin/git -C "/Users/johndoe/my-app" -c "core.editor=code --wait --reuse-window" reset --hard 9899d3cbb666bdb4e588cec6372f52b83bcfe656
zsh: no such file or directory: usr/bin/git
```

**Comment by fterradev** [comment]
And also if I have something typed in before like "foo", I get an error similar to the original one:
````
zsh: no such file or directory: foousr/bin/git
````

(Note "foo" is still there)

Even though I see those added codes to supposedly remove any text on the current line in the terminal when I first use GitLens and there was no GitLens terminal opened yet.

![image](https://github.com/gitkraken/vscode-gitlens/assets/19625575/4d0a424b-8278-437b-93b7-ecbb48af5990)

But it also fail due to having that prefix "/usr/bin/" as I said in the previous message.
````
zsh: no such file or directory: usr/bin/git
````

**Comment by eamodio** [comment]
Unfortunately I can't find a solution that works across OSs and across shells.

**Comment by farmerpaul** [comment]
This same issue started biting me yesterday, experiencing the exact same behaviour as @fterradev.  Not sure I understand why `usr/bin/git` is no longer preceded by a `/` as of the last GitLens update, but I can't really use any of its commands anymore for this reason (unless someone knows a workaround?). Thanks for investigating!

**Comment by eamodio** [comment]
Are you still seeing this in the latest pre-release? I thought I fixed this last night.

**Comment by farmerpaul** [comment]
Good call, my extension hadn't updated 🤦🏻 . Tested this morning, it's working again! Thanks!

**Comment by fterradev** [comment]
@eamodio Never mind. I consider it was mostly an error of mine as a user.

Aside from that, I don't know if it was intentional to move from using simply the `git` command, to have now the full path to the binary.
Anyway it's working fine.

**Comment by fterradev** [comment]
I don't know if you want to close as completed or close as not planned. I'll defer to you. Thanks for the support.

**Comment by eamodio** [comment]
Closing this as recent version of GitLens no longer use the terminal to execute Git commands
